tiatoolbox/models/architecture/mapde.py

```diff
@@ -0,0 +1,294 @@
+                    0.6,
+                    0.5527864,
+                    0.43431458,
+                    0.27888975,
+                    0.1055728,
+                    0.0,
+                ],
+                [
+                    0.0,
+                    0.0,
+                    0.1514719,
+                    0.27888975,
+                    0.36754447,
+                    0.4,
+                    0.36754447,
+                    0.27888975,
+                    0.1514719,
+                    0.0,
+                    0.0,
+                ],
+                [
+                    0.0,
+                    0.0,
+                    0.0,
+                    0.1055728,
+                    0.17537889,
+                    0.2,
+                    0.17537889,
+                    0.1055728,
+                    0.0,
+                    0.0,
+                    0.0,
+                ],
+                [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+            ],
+            dtype=np.float32,
+        )
+
+        dist_filter = np.expand_dims(dist_filter, axis=(0, 1))  # NCHW
+        dist_filter = np.repeat(dist_filter, repeats=num_class * 2, axis=1)
+
+        self.min_distance = min_distance
+        self.threshold_abs = threshold_abs
+        self.register_buffer(
+            "dist_filter",
+            torch.from_numpy(dist_filter.astype(np.float32)),
+        )
+        self.dist_filter.requires_grad = False
+
+    def forward(self, input_tensor: torch.Tensor) -> torch.Tensor:
+        """Logic for using layers defined in init.
+
+        This method defines how layers are used in forward operation.
+
+        Args:
+            input_tensor (torch.Tensor):
+                Input images, the tensor is in the shape of NCHW.
+
+        Returns:
+            torch.Tensor:
+                Output map for cell detection. Peak detection should be applied
+                to this output for cell detection.
+
+        """
+        logits, _, _, _ = super().forward(input_tensor)
+        out = F.conv2d(logits, self.dist_filter, padding="same")
+        return F.relu(out)
+
+    #  skipcq: PYL-W0221  # noqa: E800
+    def postproc(self, prediction_map: np.ndarray) -> np.ndarray:
+        """Post-processing script for MicroNet.
+
+        Performs peak detection and extracts coordinates in x, y format.
+
+        Args:
+            prediction_map (ndarray):
+                Input image of type numpy array.
+
+        Returns:
+            :class:`numpy.ndarray`:
+                Pixel-wise nuclear instance segmentation
+                prediction.
+
+        """
+        coordinates = peak_local_max(
+            np.squeeze(prediction_map[0], axis=2),
+            min_distance=self.min_distance,
+            threshold_abs=self.threshold_abs,
+            exclude_border=False,
+        )
+        return np.fliplr(coordinates)
+
+    @staticmethod
+    def infer_batch(model, batch_data, on_gpu):
```

Comment:
Do we need type hints here?

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/443/commits/b25b74c348f761586e459e00f51a2169b66edb8f